ci-operator/step-registry/gcp/provision/mirror-dns/gcp-provision-mirror-dns-commands.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+if [[ ! -f "${SHARED_DIR}/bastion_public_address" ]] || [[ ! -f "${SHARED_DIR}/bastion_private_address" ]]; then
+  echo "'${SHARED_DIR}/bastion_public_address' or '${SHARED_DIR}/bastion_private_address' not found, abort." && exit 1
+fi
+
+if [[ ! -f "${SHARED_DIR}/customer_vpc_subnets.yaml" ]]; then
+  echo "Lack of VPC info, abort." && exit 2
+fi
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+BASE_DOMAIN_ZONE_NAME="$(gcloud dns managed-zones list --filter "DNS_NAME=${BASE_DOMAIN}." --format json | jq -r .[0].name)"
+CLUSTER_NETWORK=$(/tmp/yq r "${SHARED_DIR}/customer_vpc_subnets.yaml" "platform.gcp.network")
+
+bastion_public_ip=$(< "${SHARED_DIR}/bastion_public_address")
+bastion_private_ip=$(< "${SHARED_DIR}/bastion_private_address")
+
+echo "Configuring public DNS for the mirror registry..."
+gcloud dns record-sets create "${CLUSTER_NAME}.mirror-registry.${BASE_DOMAIN}." \
+--rrdatas="${bastion_public_ip}" --type=A --ttl=60 --zone="${BASE_DOMAIN_ZONE_NAME}"
+cat > "${SHARED_DIR}/mirror-dns-destroy.sh" << EOF
+gcloud dns record-sets delete -q "${CLUSTER_NAME}.mirror-registry.${BASE_DOMAIN}." --type=A --zone="${BASE_DOMAIN_ZONE_NAME}"
+EOF
+
+echo "Configuring private DNS for the mirror registry..."
+gcloud dns managed-zones create "mirror-registry-private-zone" --description "Private zone for the mirror registry." --dns-name "mirror-registry.${BASE_DOMAIN}." --visibility "private" --networks "${CLUSTER_NETWORK}"
+gcloud dns record-sets create "${CLUSTER_NAME}.mirror-registry.${BASE_DOMAIN}." \
+--rrdatas="${bastion_private_ip}" --type=A --ttl=60 --zone="mirror-registry-private-zone"
+
+cat >> "${SHARED_DIR}/mirror-dns-destroy.sh" << EOF
```

Comment:
can we put all destroy commands in one place? that would be easy to read.

Reply:
ohh, the resource creation commands are followed by resource destruction.
get it, ignore my comment.